test/common.sh

```diff
@@ -0,0 +1,276 @@
+        # if not then run the tests for next task (if any)
+        [[ ! -d $runtest ]] && skipit=True
+        
+        ls ${resourcedir}/*.yaml 2>/dev/null >/dev/null || skipit=True
+
+        cat ${resourcedir}/*.yaml | grep 'tekton.dev/deprecated: \"true\"' && skipit=True
+
+         # In case if PLATFORM env variable is specified, do the tests only for matching tasks
+	    [[ -n ${PLATFORM} ]] && [[ $(grep "tekton.dev/platforms" ${resourcedir}/*.yaml 2>/dev/null) != *"${PLATFORM}"* ]] && skipit=True
+
+	    [[ -n ${skipit} ]] && continue
+
+        # Delete the tns if already exists
+        ${KUBECTL_CMD} delete ns ${tns} >/dev/null 2>/dev/null || :
+
+        # create the test namespace
+        ${KUBECTL_CMD} create namespace ${tns} >/dev/null 2>/dev/null
+
+        # create the service account appstudio-pipeline (konflux spedific requirement)
+        $KUBECTL_CMD create sa appstudio-pipeline -n ${tns}
+
+         # Install the task itself first. We can only have one YAML file
+        yaml=$(printf  ${resourcedir}/*.yaml)
+        started=$(date '+%Hh%M:%S')
+        echo "${started} STARTING: ${testname}/${version}"
+
+        # dry-run this YAML to validate and also get formatting side-effects.
+        # TODO: need to add tekton linter here
+        ${KUBECTL_CMD} -n ${tns} create -f ${yaml} --dry-run=client -o yaml >${TMPF}
+
+        [[ -f ${resourcedir}/tests/pre-apply-task-hook.sh ]] && source ${resourcedir}/tests/pre-apply-task-hook.sh
+        function_exists pre-apply-task-hook && pre-apply-task-hook
+
+        # Make sure we have deleted the content, this is in case of rerun
+        # and namespace hasn't been cleaned up or there is some Cluster*
+        # stuff, which really should not be allowed.
+        ${KUBECTL_CMD} -n ${tns} delete -f ${TMPF} >/dev/null 2>/dev/null || true
+        ${KUBECTL_CMD} -n ${tns} create -f ${TMPF}
+
+        # Install resource and run
+        for yaml in ${runtest}/*.yaml;do
+            cp ${yaml} ${TMPF}
+            [[ -f ${resourcedir}/tests/pre-apply-taskrun-hook.sh ]] && source ${resourcedir}/tests/pre-apply-taskrun-hook.sh
+            function_exists pre-apply-taskrun-hook && pre-apply-taskrun-hook
+
+            # Make sure we have deleted the content, this is in case of rerun
+            # and namespace hasn't been cleaned up or there is some Cluster*
+            # stuff, which really should not be allowed.
+            ${KUBECTL_CMD} -n ${tns} delete -f ${TMPF} >/dev/null 2>/dev/null || true
+            ${KUBECTL_CMD} -n ${tns} create -f ${TMPF}
+        done
+
+        resource_to_wait_for["$testname/${version}"]="${tns}|$started"
+
+    done
+
+    # I would refactor this to a function but bash limitation is too great, really need a rewrite the sooner
+    # the uglness to pass a hashmap to a function https://stackoverflow.com/a/17557904/145125
+    local cnt=0
+    local all_status=''
+    local reason=''
+    local maxloop=60 # 10 minutes max
+
+    set +x
```

Comment:
Why do we need `set +x` here?

Reply:
to avoid unnecessary logs being printed in the console, while we looping through to check task run status, reason and results, if something fails `show_failure()` will dump the test resources in the console.